test/lib/ufe/testObservableScene.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2020 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import os
+import sys
+import unittest
+
+import ufe
+
+class TestObserver(ufe.Observer):
+    def __init__(self):
+        self.add = 0
+        self.delete = 0
+        self.pathChange = 0
+        self.subtreeInvalidate = 0
+        self.composite = 0
+        super(TestObserver, self).__init__()
+
+    def __call__(self, notification):
+        if isinstance(notification, ufe.ObjectAdd):
+            self.add += 1
+        if isinstance(notification, ufe.ObjectDelete):
+            self.delete += 1
+        if isinstance(notification, ufe.ObjectPathChange):
+            self.pathChange += 1
+        if isinstance(notification, ufe.SubtreeInvalidate):
+            self.subtreeInvalidate += 1
+        if isinstance(notification, ufe.SceneCompositeNotification):
+            self.composite += 1
+
+    def notifications(self):
+        return [self.add, self.delete, self.pathChange, self.subtreeInvalidate, self.composite]
+
+class UFEObservableSceneTest(unittest.TestCase):
+
+    def checkNotifications(self, testObserver, listNotifications):
+        self.assertTrue(testObserver.add == listNotifications[0])
+        self.assertTrue(testObserver.delete == listNotifications[1])
+        self.assertTrue(testObserver.pathChange == listNotifications[2])
+        self.assertTrue(testObserver.subtreeInvalidate == listNotifications[3])
+        self.assertTrue(testObserver.composite == listNotifications[4])
+
+    def testObservableSelection(self):
```

Comment:
Copy-paste damage: should be testObservableScene(), or testSceneObservation(), or similar.

Reply:
addressed in: 871a903669aeceb3ac75c1d2e4d515e1a0144a69